biz.aQute.bndlib/src/aQute/bnd/plugin/jpms/JPMSModuleInfoPlugin.java

```diff
@@ -1,21 +1,34 @@
 package aQute.bnd.plugin.jpms;
 
+import static aQute.bnd.classfile.ModuleAttribute.ACC_OPEN;
 import static aQute.bnd.osgi.Constants.AUTOMATIC_MODULE_NAME;
+import static aQute.bnd.osgi.Constants.DYNAMICIMPORT_PACKAGE;
 import static aQute.bnd.osgi.Constants.INTERNAL_EXPORT_TO_MODULES_DIRECTIVE;
 import static aQute.bnd.osgi.Constants.INTERNAL_MODULE_VERSION_DIRECTIVE;
 import static aQute.bnd.osgi.Constants.INTERNAL_OPEN_TO_MODULES_DIRECTIVE;
 import static aQute.bnd.osgi.Constants.JPMS_MODULE_INFO;
+import static aQute.bnd.osgi.Constants.JPMS_MODULE_INFO_OPTIONS;
 import static aQute.bnd.osgi.Constants.MAIN_CLASS;
 import static aQute.bnd.osgi.Constants.MODULE_INFO_CLASS;
+import static aQute.bnd.osgi.Constants.PROVIDE_CAPABILITY;
+import static aQute.bnd.osgi.Constants.REQUIRE_CAPABILITY;
 import static aQute.bnd.osgi.Constants.SERVICELOADER_NAMESPACE;
 import static aQute.bnd.osgi.Constants.SERVICELOADER_REGISTER_DIRECTIVE;
 import static aQute.bnd.osgi.Constants.USES_DIRECTIVE;
+import static aQute.bnd.osgi.Descriptors.fqnToBinary;
+import static aQute.lib.env.Header.removeDuplicateMarker;
```

Comment:
I think here you should use `Processor.removeDuplicateMarker`.

Reply:
right